File: C++/Count_Primes.cpp
#include <iostream>
#include <vector>

using namespace std;
int countPrimes(int n) {
        vector<bool>isPrime(n + 1, true);
        isPrime[0] = isPrime[1] = false;
        
        for(int i = 2; i*i <= n; i++) {
            if(!isPrime[i]) continue;
            for(int j = i*i; j <= n; j += i) {
                isPrime[j] = false;
            }
        }
        
        int count = 0;
        
        for(int i = 2; i < n; i++) {
            if(isPrime[i]) count++;
        }
        return count;
    }
    
int main(){
    int n;
    cin>>n;
    countPrimes(n);
    cout<<countPrimes(n);
    return 0;
};

File: Python/Count_Primes.py
# https://leetcode.com/problems/count-primes/

class Solution:
    def countPrimes(self, n: int) -> int:
        prime = [True for i in range(n+1)]
        p = 2
        while (p * p <= n):
            if (prime[p] == True):
                for i in range(p * p, n+1, p):
                    prime[i] = False
            p += 1
        sol = 0
        for p in range(2, n):
            if prime[p]:
                sol += 1
        return sol
            